internal/worker/clienterrors/errors.go

```diff
@@ -49,15 +50,35 @@ const (
 type ClientErrorCode int
 
 type Error struct {
-	ID      ClientErrorCode `json:"id"`
-	Reason  string          `json:"reason"`
-	Details interface{}     `json:"details,omitempty"`
+	ID      ClientErrorCode
+	Reason  string
+	Details interface{}
```

Comment:
why did you remove the `json:..` here? isn't that needed for decoding?

Reply:
Probably because I'm a Muppet (sometimes at least :) - if we also decode I need to re-add it, it is a bit annoying that it's duplicated but I am not sure there is a nicer way with go, I will look though.